packages/client/hmi-client/src/components/workflow/ops/model/tera-model-drilldown.vue

```diff
@@ -5,21 +5,35 @@
 		@update-state="(state: any) => emit('update-state', state)"
 	>
 		<tera-drilldown-section>
-			<tera-model v-if="node.state.modelId" :asset-id="node.state.modelId" />
+			<tera-model v-if="node.state.modelId" :asset-id="node.state.modelId" @on-save="onSaveEvent" />
 		</tera-drilldown-section>
 	</tera-drilldown>
 </template>
 
 <script setup lang="ts">
+import _ from 'lodash';
 import { WorkflowNode } from '@/types/workflow';
-import { ModelOperationState } from '@/components/workflow/ops/model/model-operation';
+import { ModelOperation, ModelOperationState } from '@/components/workflow/ops/model/model-operation';
 import TeraModel from '@/components/model/tera-model.vue';
 import TeraDrilldown from '@/components/drilldown/tera-drilldown.vue';
 import TeraDrilldownSection from '@/components/drilldown/tera-drilldown-section.vue';
 
-defineProps<{
+const props = defineProps<{
 	node: WorkflowNode<ModelOperationState>;
 }>();
 
-const emit = defineEmits(['close', 'update-state']);
+const emit = defineEmits(['close', 'update-state', 'append-output']);
+
+const onSaveEvent = (event: any) => {
+	const state = _.cloneDeep(props.node.state);
+	state.modelId = event.id;
+	emit('update-state', state);
+	emit('append-output', {
+		type: ModelOperation.outputs[0].type,
+		label: event.header.name,
+		value: [event.id],
+		state,
+		isSelected: false
+	});
+};
```

Comment:
do we want to append an output or do we want to replace it?

Reply:
We replacing it, will fix this.

Reply:
You want to append, we don't want to replace.

Reply:
I'm editing this PR to do that. We need to be able to edit current output. Then the goal is to also propagate changes if the ouput has been updated.